lib/cc/engine/analyzers/violation.rb

```diff
@@ -57,7 +61,15 @@ def name
         end
 
         def calculate_points
-          base_points * issue.mass
+          if issue.mass > threshold
```

Comment:
That's partly why I pointed it out: `DEFAULT_POINTS` *happens* to be correct right now, but only because `BASE_POINTS` is the same for all the languages. Since that could change, depending upon it would basically be a bug-in-waiting.

As @GordonDiggs suggested above, if we could define those values in the `Base` impl, that would certainly help clarify things.